activerecord/lib/arel/nodes/select_core.rb

```diff
@@ -55,7 +57,8 @@ def eql?(other)
           self.wheres == other.wheres &&
           self.groups == other.groups &&
           self.havings == other.havings &&
-          self.windows == other.windows
+          self.windows == other.windows &&
+          self.comment == other.comment
       end
       alias :== :eql?
     end
```

Comment:
Suggests there's a single comment per statement, when there's a single annotation that is a comment node.

What would we call this if we had other comments in the statement? Annotation as well?